jax/numpy/lax_numpy.py

```diff
@@ -823,9 +823,10 @@ def histogram_bin_edges(a, bins=10, range=None, weights=None):
   if b.ndim == 1:
     return b
   if range is None:
-    range = a.min(), a.max()
-  if not isinstance(a, core.Tracer) and range[0] == range[1]:
-    range = range[0] - 0.5, range[0] + 0.5
+    range = (where(a.ptp() == 0, a.min() - 0.5, a.min()),
+             where(a.ptp() == 0, a.max() + 0.5, a.max()))
+  elif range[0] == range[1]:
```

Comment:
Implicitly this assumes that `range` is known at compile time. Do we want to support non-compile-time `range`s?

Reply:
Hmm... I'd been assuming range is a compile-time constant, but I suppose it's easy enough to accommodate. I'll do that.

Reply:
Done.